src/main/java/persistence/entity/DirtyCheck.java

```diff
@@ -0,0 +1,25 @@
+package persistence.entity;
+
+import persistence.sql.dml.ColumnValues;
+
+public class DirtyCheck {
+
+    private final PersistenceContext persistenceContext;
+
+    public DirtyCheck(PersistenceContext persistenceContext) {
+        this.persistenceContext = persistenceContext;
+    }
+
+    public boolean isDirty(Object entity) throws IllegalAccessException {
+        Object snapshot = persistenceContext.getDatabaseSnapshot(entity);
+        return findDirty(new ColumnValues<>(snapshot), new ColumnValues<>(entity));
+    }
+
+    private boolean findDirty(ColumnValues previousColumnValues, ColumnValues currentColumnValues) {
```

Comment:
```suggestion
    private <T> boolean findDirty(ColumnValues<T> previousColumnValues, ColumnValues<T> currentColumnValues) {
```
제네릭을 활용한 만큼 표기를 명확하게 해주는 노력도 필요하겠네요~